esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5999,24 +5997,36 @@ void CWsDfuEx::getFileMeta(StringBuffer &metaInfoStr, StringBuffer &expiryTime,
     JBASE64_Encode(compressedMetaInfoMb.bytes(), compressedMetaInfoMb.length(), metaInfoStr, false);
 }
 
+void CWsDfuEx::getFileDafilesrvConfiguration(StringBuffer &keyPairName, unsigned &retPort, bool &retSecure, const char *cluster)
+{
+    retPort = DEFAULT_ROWSERVICE_PORT;
+    retSecure = false;
+    const char *_keyPairName = env->getClusterKeyPairName(cluster);
+    Owned<IConstDaFileSrvInfo> daFileSrvInfo = env->getDaFileSrvGroupInfo(cluster);
+    unsigned port = DEFAULT_ROWSERVICE_PORT;
+    bool secure = false;
+    if (daFileSrvInfo)
+    {
+        port = daFileSrvInfo->getPort();
+        secure = daFileSrvInfo->getSecure();
+    }
```

Comment:
@wangkx - I see you copy and pasted my pseudo code :)

The above doesn't set any of of the the return params!

Reply:
This might be more accurate, but please check it's functionality.
```
void CWsDfuEx::getFileDafilesrvConfiguration(StringBuffer &keyPairName, unsigned &port, bool &secure, const char *cluster)
{
    port = DEFAULT_ROWSERVICE_PORT;
    secure = false;
    keyPairName.set(env->getClusterKeyPairName(cluster));
    Owned<IConstDaFileSrvInfo> daFileSrvInfo = env->getDaFileSrvGroupInfo(cluster);
    if (daFileSrvInfo)
    {
        port = daFileSrvInfo->getPort();
        secure = daFileSrvInfo->getSecure();
    }
}
```